src/extractor.rs

```diff
@@ -157,6 +147,73 @@ impl Arbitrary for ConfigVal {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct DbtRef {
+    pub name: String,
+    pub package: Option<String>,
+    pub version: Option<RefVersion>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum RefVersion {
+    StringRV(String),
+    IntRV(i64),
+    DoubleRV(f64),
+}
+
+#[cfg(test)]
+impl Arbitrary for RefVersion {
+    fn arbitrary(g: &mut Gen) -> RefVersion {
+        let kind: usize = usize::arbitrary(g) % 2;
+
+        match kind {
+            0 => RefVersion::StringRV(String::arbitrary(g)),
```

Comment:
[nit] you could put `use RefVersion::*` at the top so you don't have to use the fully qualified constructors everywhere. They're named in such a way they won't collide.

Reply:
> They're named in such a way they won't collide
This gives me the heebbie jeebbies. I've been burned so many times by relying on naming conventions versus using a namespacing mechanic. I would choose the namespacing mechanic eight days a week; but that's just my opinion.

Reply:
Rust is so pedantic about types that it would probably be hard to be burned.

Reply:
Fair worry, but the Rust's name mangling strategy retains both fully qualified packages as well as versions even within the same package (so you don't have to use shadowing to use two versions of the same package). Testing locally now creating my own `Vec` type you get the following error:
> expected struct `Vec`, found struct `std::vec::Vec`